### ISSUE TITLE
Documentation problem: Installing & Running khmer

### ISSUE THREAD

**Issue by ACharbonneau** [dib-lab/khmer#1156]
I believe that I have the listed requirements (http://khmer.readthedocs.org/en/latest/user/install.html):
Python 2.7.6 (default, Sep  9 2014, 15:04:36)
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)] on darwin

but I can't get the "Run the tests" section to work.

I get:
$ nosetests khmer --attr '!known_failing'
>-bash: nosetests: command not found

$ pip2 install khmer[tests]
>Requirement already satisfied (use --upgrade to upgrade): khmer[tests] in ./khmerEnv/lib/python2.7/site-packages
 >Installing extra requirements: 'tests'
>Cleaning up...

$ nosetests khmer --attr '!known_failing'
>-bash: nosetests: command not found

$ pip2 install --upgrade khmer[tests]
>Requirement already up-to-date: khmer[tests] in ./khmerEnv/lib/python2.7/site-packages
  Installing extra requirements: 'tests'
>Cleaning up...

$ nosetests khmer --attr '!known_failing'
>-bash: nosetests: command not found

**Comment by ctb** [comment]
On Tue, Jul 07, 2015 at 11:36:51AM -0700, Amanda Charbonneau wrote:
> I believe that I have the listed requirements (http://khmer.readthedocs.org/en/latest/user/install.html):

Try 'pip2 install nose' and let us know if that works - thanks!

best,
--titus

**Comment by luizirber** [comment]
Sometimes the PATH is not updated, try to `deactivate` and `source khmerEnv/bin/activate` again and `nosetests` should be available.

**Comment by ACharbonneau** [comment]
deactivated/reactivated as per Luiz; did 'pip2 install nose' as per Titus...not sure if this is the expected result:

$ nosetests khmer --attr '!known_failing'
>.....Killed: 9

**Comment by luizirber** [comment]
Can you check if
`$ nosetests khmer --attr '!known_failing,!huge'`
works? If so, we probably need to update the instructions...
(I think the `huge` tests don't work on OSX)

**Comment by ACharbonneau** [comment]
I told Crusoe I would field test all of the documentation :) 
And I am on a Mac. So:

$ nosetests khmer --attr '!known_failing,!huge'
>.....Killed: 9

With an extra space:
$ nosetests khmer --attr '!known_failing, !huge'

>----------------------------------------------------------------------
>Ran 0 tests in 0.006s

>OK

Expected result?

**Comment by bocajnotnef** [comment]
@ACharbonneau Try `nosetests tests --attr '!known_failing, !huge'`, please.

**Comment by ACharbonneau** [comment]
Sorry, I can't get this to format in a useful way, so I also attached an image:

>$ nosetests tests --attr '!known_failing, !huge'

E
======================================================================
ERROR: Failure: ImportError (No module named tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/Amanda/testKhmer/khmerEnv/lib/python2.7/site-packages/nose/loader.py", line 407, in loadTestsFromName
    module = resolve_name(addr.module)
  File "/Users/Amanda/testKhmer/khmerEnv/lib/python2.7/site-packages/nose/util.py", line 312, in resolve_name
    module = __import__('.'.join(parts_copy))
ImportError: No module named tests

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)

![image](https://cloud.githubusercontent.com/assets/1719360/8605982/c6d634ca-2656-11e5-996a-b7682a6c1d64.png)

**Comment by bocajnotnef** [comment]
Hrm. That's odd. What directory are you in? khmer's root?

**Comment by luizirber** [comment]
This is installed through pip, @bocajnotnef , so there is no khmer's root.

**Comment by bocajnotnef** [comment]
Ahh.

**Comment by ACharbonneau** [comment]
I'm assuming this means that the nosetests still aren't working. 
Does this mean that khmer is likely installed incorrectly? Or just that the testing modules are?
And is there an alternative way to make sure that khmer is installed as expected? 

I don't want to attempt to test subsequent documentation with a borked install.

**Comment by mr-c** [comment]
@ACharbonneau Thank you for testing!

Can I see the output of
```
echo $PWD
echo $PATH
ls
pip show khmer
pip show nose
```

**Comment by ACharbonneau** [comment]
Hey Michael! The instructions suggest that I should be running the tests from the main directory, and that's where I ran these, but I've also attempted all the tests from inside the khmerEnv directory with the same results. ls of khmerEnv is appended

$ echo $PWD
>/Users/Amanda/testKhmer

$ echo $PATH
>/Users/Amanda/testKhmer/khmerEnv/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin

$ ls
>khmerEnv			virtualenv-1.11.6		virtualenv-1.11.6.tar.gz

$ pip show khmer
>Name: khmer
>Version: 1.4.1
>Location: /Users/Amanda/testKhmer/khmerEnv/lib/python2.7/site-packages
>Requires: screed

$ pip show nose
>Name: nose
>Version: 1.3.7
>Location: /Users/Amanda/testKhmer/khmerEnv/lib/python2.7/site-packages

$ cd khmerEnv/ ; ls
>bin	include	lib	man

**Comment by mr-c** [comment]
Huh, looks like you have a valid installation of khmer and that your virtualenv works.

Running `nosetests khmer --attr '!known_failing,!linux'` from any directory while the virtualenv is active should work (as long as there isn't a directory named 'khmer' inside that directory).

Here is the expected output:

```
(khmerEnv)colin-think:~ mcrusoe$ nosetests khmer --attr '!known_failing,!linux'
.........................................................................................................ERROR: Input file /var/folders/_s/hs5cgvld491g2m1pjvgqlxhm0000gn/T/khmertest_s_KzMY/thisfiledoesnotexist does not exist
Exiting
.ERROR: Input file /var/folders/_s/hs5cgvld491g2m1pjvgqlxhm0000gn/T/khmertest_nLfeUJ/thisfiledoesnotexist does not exist
..................................................................................................................................................................................................................SSSSS...................................................................................................................................................................
----------------------------------------------------------------------
Ran 484 tests in 21.869s

OK (SKIP=5)
```

**Comment by mr-c** [comment]
@ACharbonneau Would you like to debug this interactively over Google Hangouts some time?

**Comment by ACharbonneau** [comment]
That would be great if you have the time @mr-c. Otherwise I was going to see if I could get @ctb to look at it during NGS. I can make time anytime next week except Monday from 4-6 and Tuesday 3-5 (EST)

**Comment by mr-c** [comment]
7PM EST on Monday or 6PM EST on Tuesday works for me

**Comment by ACharbonneau** [comment]
Monday at 7 then.

**Comment by ACharbonneau** [comment]
nosetests tests --attr '!known_failing, !huge, !linux'

#fixed

**Comment by mr-c** [comment]
Fixed in #1247
